src/components/project-list.js

```diff
@@ -1,21 +1,48 @@
-import React, {useState } from 'react';
+import React, { useState, useEffect, useContext,  } from 'react';
+import axios from 'axios'; 
 import Table from './table';
 import Project from './project';
-
+import { useUserContext } from './user-context';
 
 function ProjectList() {
-  const [projects, setProjects] = useState([
-    { name: 'project 1', hidden: false, isStarred: false },
-    { name: 'project 2', hidden: false, isStarred: false },
-    { name: 'project 3', hidden: false, isStarred: false },
-    { name: 'project 4', hidden: false, isStarred: false },
-    { name: 'project 5', hidden: false, isStarred: false },
-    { name: 'project 6', hidden: false, isStarred: false },
-    { name: 'project 7', hidden: false, isStarred: false },
-    { name: 'project 8', hidden: false, isStarred: false }
-  ]);
+  const [projects, setProjects] = useState([]);
+  const token = useUserContext();
+
+  const axiosInstance = axios.create({
+    baseURL: 'https://sheetlabs.com/ARBI',
+  });
+
+  axiosInstance.interceptors.request.use(
```

Comment:
@rykabdullah7 we need to create this interceptor globally in a new file. The interceptor should add token to each request and if a call returns with  401 status code, then it should send the user to login page.